plugins/hls-overloaded-record-dot-plugin/src/Ide/Plugin/OverloadedRecordDot.hs

```diff
@@ -105,7 +117,14 @@ instance Hashable CollectRecordSelectors
 instance NFData CollectRecordSelectors
 
 data CollectRecordSelectorsResult = CRSR
-    { recordInfos       :: RangeMap RecordSelectorExpr
+    { -- |We store everything in here that we need to create the unresolved
+      -- codeAction, the range, an uniquely identifiable int, and the expression
+      -- for the selector expression that we use to generate the name
+      records           :: RangeMap (Int, HsExpr (GhcPass 'Renamed))
```

Comment:
Why are we not using the actual `Unique` as the key?

Reply:
Because we need to serialize the Unique when we put it in the data field, and it doesn't have json instances. I could make orphans for it, but that didn't seem worth it when it's just a wrapper around an integer itself.

Reply:
Hmm, that's a bit awkward. Worth a comment explaining that, since in theory `Data.Unique` suggests you should only really use it as a hash key.